src/treetensornetwork/projttno.jl

```diff
@@ -0,0 +1,52 @@
+"""
+ProjTTNO
+"""
+mutable struct ProjTTNO <: AbstractProjTTNO
+  pos::Union{Vector{<:Tuple},NamedDimEdge{Tuple}} # TODO: cleanest way to specify effective Hamiltonian position?
+  H::TTNO
+  environments::Dictionary{NamedDimEdge{Tuple},ITensor}
```

Comment:
It may be nice to make this a directed `DataGraph`.

Reply:
Could be a good idea, adding/invalidating environments by adding/removing directed edges sounds nice. I would have to think about what to do with the inside the operator position for a 2-site effective operator (remove them altogether?) and whether I could get rid of some other things by using more `DataGraph`-like methods instead.